cnapy/gui_elements/config_cobrapy_dialog.py

```diff
@@ -0,0 +1,133 @@
+"""The COBRApy configuration dialog"""
+import configparser
+import os
+import appdirs
+
+from cobra.util.solver import interface_to_str, solvers
+from cobra import Configuration
+from multiprocessing import cpu_count
+
+from qtpy.QtGui import QDoubleValidator, QIntValidator
+from qtpy.QtWidgets import (QMessageBox, QComboBox, QDialog,
+                            QHBoxLayout, QLabel, QLineEdit, QPushButton,
+                            QVBoxLayout)
+import cnapy.resources
+from cnapy.cnadata import CnaData
+
+class ConfigCobrapyDialog(QDialog):
+    """A dialog to set values in cobrapy-config.txt"""
+
+    def __init__(self, appdata: CnaData):
+        QDialog.__init__(self)
+        self.setWindowTitle("Configure COBRApy")
+
+        self.appdata = appdata
+        self.layout = QVBoxLayout()
+
+        # allow MILP solvers only?
+        avail_solvers = list(set(solvers.keys()) - {'scipy'}) # SCIPY currently not even usable for FBA
+
+        h2 = QHBoxLayout()
+        label = QLabel("Default solver:\n(set when loading a model)")
+        h2.addWidget(label)
+        self.default_solver = QComboBox()
+        self.default_solver.addItems(avail_solvers)
+        self.default_solver.setCurrentIndex(avail_solvers.index(interface_to_str(Configuration().solver)))
```

Comment:
I would only `import cobra` instead of `from cobra import Configuration` 
Then i would once do a `cobra_config = cobra.Configuration()` and then use `cobra_config` everywhere where now `Configuration()` is used.

Reply:
Configuration() is a singleton class so by using it directly I think it becomes clearer that this is a global configuration.

Reply:
I see it like that. If I look at the code locally I don't know where `Configuration()` comes from I have to see the import on top for that. It's definitely a matter of taste but it looks strange to me. I tend to use qualification like `cobra.Configuration()`  to avoid name collisions. This might be not be relevant here but `Configuration` is probably a ubiquitous term. :shrug: